tensorflow/python/ops/image_grad_test_base.py

```diff
@@ -535,17 +536,28 @@ def crop_resize(image_tensor, boxes_tensor):
               with test_util.device(use_gpu=True):
                 with self.cached_session():
                   # pylint: disable=cell-var-from-loop
-                  err1 = gradient_checker_v2.max_error(
-                      *gradient_checker_v2.compute_gradient(
+                  if (self.__class__.__name__ ==
+                      "CropAndResizeOpDeterministicTest" and
```

Comment:
Instead of checking for the class name, I would check if `os.environ['TF_DETERMINISTIC_OPS']` is "1".

Also, maybe just disable this test when determinism is used, since you already test the backprop error messages elsewhere.

Reply:
I chose to do it this way to document the provenance of this conditional branch (i.e. in a child class) and to encapsulate the mechanism by which op-determinism is enabled. However, I don't see any major downsides to changing this, specifically:

  1. if this condition is removed by someone who doesn't understand why it's there or why it's needed then tests will fail and the person who made the change will be alerted to the error, and
  2. when `TF_DETERMINISTIC_OPS` is replaced with `tf.config.*deterministic*`, and `TF_DETERMINISTIC_OPS` is no longer set to '1' in the child class, then this test will fail because the exceptions will no longer be thrown.

> Also, maybe just disable this test when determinism is used, since you already test the backprop error messages elsewhere.

Testing that the functionality in this conditional branch is correct will ensure that when the exceptions are removed and deterministic GPU backprop functionality is added, this test will be modified to test that functionality. If the gradient is just skipped when determinism is expected then there is a higher probability that the deterministic GPU backprop functionality will (accidentally) not be tested.

This will be changed to `os.getenv('TF_DETERMINISTIC_OPS', '0') == '1'` in the next commit.